### PR TITLE
Sleeping in Class solution

### PR DESCRIPTION
**Pull request by yiminghuang47**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making the changes.

### CODE REVIEW

File: solutions/usaco-1203.mdx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+title: Sleeping in Class
+author: Yiming Huang
+---
+
+[Official Analysis](http://usaco.org/current/data/sol_prob1_bronze_feb22.html)
+
+
+## Explanation
+
+After Elsie has made all the modifications to the log array $A[]$, the log will show that Bessie sleeps the same number of hours every day. Let's call that number $\texttt{numHours}$.
```
Comment (SansPapyrus683) [inline solutions/usaco-1203.mdx:13]: just say A, not A[]

File: solutions/usaco-1203.mdx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+			return false;  // currsum cannot exceed x
+		}
+		else if (currsum == numHours) {
+			currsum = 0;  // reset currsum
+		}
+	}
+	return true;
+}
+
+void solve() {
```
Comment (SansPapyrus683) [inline solutions/usaco-1203.mdx:53]: just put this in main

File: solutions/usaco-1203.mdx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+	A = vector<int>(N);
+	int sum = 0; // sum of all values in the vector
+	for (int i = 0; i < N; i++)
+	{
+		cin >> A[i];
+		sum += A[i];
+	}
+
+	// numHours is the # of hours of sleep every class AFTER modification
+	// iterate over numHours from small to large because the smallest possible numHours is optimal
```
Comment (SansPapyrus683) [inline solutions/usaco-1203.mdx:65]: over 80 chars

File: solutions/usaco-1203.mdx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int N;
+vector<int> A;
+
+bool check(int numHours) {
```
Comment (SansPapyrus683) [inline solutions/usaco-1203.mdx:39]: what this function do

File: solutions/usaco-1203.mdx [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+	}
+
+	// numHours is the # of hours of sleep every class AFTER modification
+	// iterate over numHours from small to large because the smallest possible numHours is optimal
+	for (int numHours = 0; numHours <= sum; numHours++) {
+		if (numHours != 0 && sum % numHours != 0) {
+			// sum must be divisible by numHours
+			continue;
+		}
+		if (check(numHours)) {
```
Comment (SansPapyrus683) [inline solutions/usaco-1203.mdx:71]: use snake_case for c++

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

**Comment by yiminghuang47** [comment]
Sorry it's my first time adding solution.I might have undone some changes you made.

**Comment by yiminghuang47** [comment]
I did not realize that you already made changes lol. my bad

File: solutions/usaco-1203.mdx [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+		A = vector<int>(N);
+		int sum = 0; // sum of all values in the vector
+		for (int i = 0; i < N; i++) {
+			cin >> A[i];
+			sum += A[i];
+		}
+
+		// num_hours is the # of hours of sleep every class AFTER modification
+		// iterate over num_hours from small to large
+		// because the smallest possible num_hours is optimal
```
Comment (SansPapyrus683) [inline solutions/usaco-1203.mdx:81]:
```cpp
/* 
 * multiline comments should be
 * formatted like this
 */
```

File: solutions/usaco-1203.mdx [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+		cin >> N;
+		A = vector<int>(N);
+		int sum = 0; // sum of all values in the vector
+		for (int i = 0; i < N; i++) {
+			cin >> A[i];
+			sum += A[i];
+		}
+
+		/*
+		 *num_hours is the # of hours of sleep every class AFTER modification
```
Comment (SansPapyrus683) [inline solutions/usaco-1203.mdx:80]: space after asterisk

File: solutions/usaco-1203.mdx [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int N;
+vector<int> A;
+
+/** @return if num_hours leads to a valid answer */
+bool check(int num_hours) {
+	int curr_sum = 0; // keeps track of the current sum
```
Comment (SansPapyrus683) [inline solutions/usaco-1203.mdx:54]: two spaces after line of code before beginning an inline comment

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

**Review by SansPapyrus683 — APPROVED** [review]
@envyaims

File: solutions/usaco-1203.mdx [hunk 9]
```diff
@@ -0,0 +1,115 @@
+---
+id: usaco-1203
+source: USACO bronze 2022 February
```
Comment (SansPapyrus683) [inline solutions/usaco-1203.mdx:3]:
```suggestion
source: USACO Bronze 2022 February
```

Comment (SansPapyrus683) [inline solutions/usaco-1203.mdx:3]:
```suggestion
source: USACO Bronze 2022 February
```

File: solutions/usaco-1203.mdx [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+$\texttt{sum}$.
+
+Therefore, the length of the modified log will be
+$\texttt{sum}/\texttt{num\_hours}$. Since every modification reduces the length
+of the log by 1, the number of modifications will be
+$N-\texttt{sum}/\texttt{num\_hours}$.
+
+$N$ and $\texttt{sum}$ are constant, so in order to minimize
+$N-\texttt{sum}/\texttt{num\_hours}$, we need to minimize $\texttt{num\_hours}$
+too.
```
Comment (ChuangSheep) [inline solutions/usaco-1203.mdx:26]:
```suggestion
$N$ and $\texttt{sum}$ are constant, so, to minimize $N-\texttt{sum}/\texttt{num\_hours}$, 
we need to minimize $\texttt{num\_hours}$ too.
```

File: solutions/usaco-1203.mdx [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+
+In order to do that, we iterate over possible values of $\texttt{num\_hours}$ and
+checks if it's valid. If it is, then we print out the answer
+$N-\texttt{sum}/\texttt{num\_hours}$ and return the program.
+
+In order to test if a value of $\texttt{num\_hours}$ is valid, we iterate over
+the $A$ from left to right and combine values. If at some point a combined
+value exceed $\texttt{num\_hours}$ then we know that $\texttt{num\_hours}$ is not
+valid.
```
Comment (ChuangSheep) [inline solutions/usaco-1203.mdx:35]:
```suggestion
We iterate over $A$ from left to right and combine values to test if a value 
of $\texttt{num\_hours}$ is valid. If at some point a combined value exceeds 
$\texttt{num\_hours}$ then we know that $\texttt{num\_hours}$ is not
valid.  
```

**Review by ChuangSheep — COMMENTED** [review]
looks good

File: solutions/usaco-1203.mdx [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+	return true;
+}
+
+int main() {
+	int T;
+	cin >> T;
+	for (int t = 0; t < T; t++) {
+		cin >> N;
+		A = vector<int>(N);
+		int sum = 0;  // sum of all values in the vector
```
Comment (envyaims) [inline solutions/usaco-1203.mdx:73]:
```suggestion
		int sum = 0;  
```
a bit extraneous comment

File: solutions/usaco-1203.mdx [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+		 * iterate over num_hours from small to large
+		 * because the smallest possible num_hours is optimal
+		 */
+		for (int num_hours = 0; num_hours <= sum; num_hours++) {
+			if (num_hours != 0 && sum % num_hours != 0) {
+				// sum must be divisible by num_hours
+				continue;
+			}
+			if (check(num_hours)) {
+				// check(num_hours) checks if num_hours is a valid answer
```
Comment (envyaims) [inline solutions/usaco-1203.mdx:90]:
```suggestion
```

**Review by envyaims — APPROVED** [review]
gr8

**Review by ChuangSheep — APPROVED** [review]
nice

**Pull request merged**